[Python] Crash with nested imports

Given a program with the following structure:
```bash
.
├── ex.py
└── l
    ├── __init__.py
    └── md.py

1 directory, 3 files
```
with the following file contents:
`ex.py`:
```python
import l
from l import Foo

f: Foo = l.create('foo')
```
` l/__init__.py`:
```python3
from typing import Any
import l.md

class Foo:
    def __init__(self) -> None:
        self._md: md.Bar = md.Bar('bar')

def create(s: str) -> Any:
    if s == "foo":
        return Foo()
    else:
        assert False, "Invalid string"
```
`l/md.py`:
```python3 
class Bar:
    def __init__(self, s: str) -> None:
        self.s = s

    def bar(self) -> str:
        return self.s
```
ESBMC crashes as follows:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
l/__init__.py:6: error: Name "md" is not defined  [name-defined]
Found 1 error in 1 file (checked 1 source file)

ERROR: Module 'l.Foo' not found.
Please install it with: pip3 install l.Foo
Converting
terminate called after throwing an instance of 'nlohmann::json_abi_v3_11_3::detail::type_error'
  what():  [json.exception.type_error.302] type must be string, but is number
Aborted (core dumped)
```

--- a/regression/python/github_2931/l/__init__.py
+++ b/regression/python/github_2931/l/__init__.py
@@ -1,0 +1,12 @@
+from typing import Any
+import l.md as md
+
+class Foo:
+    def __init__(self) -> None:
+        self._md: md.Bar = md.Bar('bar')
+
+def create(s: str) -> Any:
+    if s == "foo":
+        return Foo()
+    else:
+        assert False, "Invalid string"

--- a/regression/python/github_2931/l/md.py
+++ b/regression/python/github_2931/l/md.py
@@ -1,0 +1,6 @@
+class Bar:
+    def __init__(self, s: str) -> None:
+        self.s = s
+
+    def bar(self) -> str:
+        return self.s

--- a/regression/python/github_2931/main.py
+++ b/regression/python/github_2931/main.py
@@ -1,0 +1,4 @@
+import l
+from l import Foo
+
+f: Foo = l.create('foo')

--- a/regression/python/github_2931/test.desc
+++ b/regression/python/github_2931/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+
+^VERIFICATION SUCCESSFUL$

--- a/src/python-frontend/python_converter.cpp
+++ b/src/python-frontend/python_converter.cpp
@@ -4868,17 +4868,29 @@ void python_converter::get_attributes_from_self(
     {
       const std::string &attr_name = stmt["target"]["attr"];
 
-      // Check if "id" exists before accessing it
-      if (!stmt["annotation"].contains("id"))
+      // Handle both simple names (id) and module-qualified names (Attribute)
+      std::string annotated_type;
+
+      if (stmt["annotation"].contains("id"))
+      {
+        // Simple type annotation such as self._md: Bar
+        annotated_type = stmt["annotation"]["id"].get<std::string>();
+      }
+      else if (
+        stmt["annotation"].contains("_type") &&
+        stmt["annotation"]["_type"] == "Attribute")
+      {
+        // Module-qualified type annotation like: self._md: md.Bar
+        // Extract just the class name (the attribute part)
+        annotated_type = stmt["annotation"]["attr"].get<std::string>();
+      }
+      else
       {
         log_warning(
           "Skipping attribute '{}' with unsupported annotation type",
           attr_name);
         continue;
       }
-
-      const std::string &annotated_type =
-        stmt["annotation"]["id"].get<std::string>();
 
       typet type;
       if (annotated_type == "str")
@@ -4889,8 +4901,7 @@ void python_converter::get_attributes_from_self(
         type = gen_pointer_type(base_type);
       }
       else
-        type =
-          type_handler_.get_typet(stmt["annotation"]["id"].get<std::string>());
+        type = type_handler_.get_typet(annotated_type);
 
       struct_typet::componentt comp =
         build_component(current_class_name_, attr_name, type);

